### PR TITLE
fix eval metric during training and add metric unittest

### PR DESCRIPTION
**Pull request by Cathy0908**
1. fix bug: `dataset.visualize` modified values of results['detection_classes'] from class id to class names , which cause the wrong eval metric values during the training
2. support visualization for oss image file
3. add train and eval metric unittest

### CODE REVIEW

File: easycv/file/image.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                                        ], 'Only support `RGB` and `BGR` mode!'
+                assert img is not None
+                break
+        except Exception as e:
+            logging.error(e)
+            logging.warning('Read file {} fault, try count : {}'.format(
+                img_path, try_cnt))
+            # frequent access to oss will cause error, sleep can aviod it
+            if is_oss_path(img_path):
+                time.sleep(1)
```
Comment (wenmengzhou) [inline easycv/file/image.py:36]: add logging after n-trys incase this tricky behaviour will cause some invisble hang

File: easycv/file/image.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        try_cnt += 1
+
+    if img is None:
+        raise ValueError('Read Image Times Out: ' + img_path)
```
Comment (wenmengzhou) [inline easycv/file/image.py:40]: Read Image Error

File: easycv/datasets/detection/raw.py [hunk 3]
```diff
@@ -114,15 +115,15 @@ def visualize(self, results, vis_num=10, score_thr=0.3, **kwargs):
                     detection_classes.append(None)
                 else:
                     detection_classes.append(
-                        np.array([class_names[id] for id in classes_id]))
-            results['detection_classes'] = detection_classes
+                        np.array([class_names[int(id)] for id in classes_id]))
+        else:
+            detection_classes = results.get('detection_classes', [])
```
Comment (wenmengzhou) [inline easycv/datasets/detection/raw.py:120]: add some comment when we need to convert class_id to class name, and when not

**Review by wenmengzhou — CHANGES_REQUESTED** [review]
minor change is needed

File: easycv/file/image.py (hunk 2 above)
Comment (Cathy0908) [reply easycv/file/image.py]: done!

File: easycv/datasets/detection/raw.py (hunk 3 above)
Comment (Cathy0908) [reply easycv/datasets/detection/raw.py:122]: done!

File: easycv/datasets/detection/raw.py [hunk 4]
```diff
@@ -107,22 +108,24 @@ def visualize(self, results, vis_num=10, score_thr=0.3, **kwargs):
         elif hasattr(self.data_source, 'classes'):
             class_names = self.data_source.classes
 
+        # If class_names is not None, class_id will be converted to class_name for visualization,
+        # otherwise the class_id will be displayed.
```
Comment (wenmengzhou) [inline easycv/datasets/detection/raw.py:112]: I mean when should user provide class_names and when not, detection_classes has different meaning  when class_names exists

**Review by wenmengzhou — COMMENTED** [review]
some comment

File: easycv/datasets/detection/raw.py (hunk 4 above)
Comment (Cathy0908) [reply easycv/datasets/detection/raw.py:112]: It’s a auto conversion and only be done during visualization. In fact, user does not need to care, and detection_classes here is just a temporary variable

Comment (wenmengzhou) [reply easycv/datasets/detection/raw.py:112]: this point should be made clear for our self-maintainance

**Review by wenmengzhou — APPROVED** [review]

**Pull request merged**